sympy/polys/tests/test_polytools.py

```diff
@@ -3933,3 +3936,167 @@ def test_issue_20985():
     w, R = symbols('w R')
     poly = Poly(1.0 + I*w/R, w, 1/R)
     assert poly.degree() == S(1)
+
+
+
+
+def test_factor_system():
+
+    def assert_equal_unordered(actual, expected):
+        assert len(actual) == len(expected)
+        actual_set = {frozenset(s) for s in actual}
+        expected_set = {frozenset(s) for s in expected}
+        assert actual_set == expected_set
+
+    assert_equal_unordered(factor_system([x**2 + 2*x + 1]), [{x + 1}])
+    assert_equal_unordered(factor_system([x**2 + 2*x + 1, y**2 + 2*y + 1]), [{x + 1, y + 1}])
+
+    assert_equal_unordered(factor_system([x**2 + 1]), [{x**2 + 1}])
+
+    assert_equal_unordered(factor_system([]), [])
+    assert_equal_unordered(factor_system([x**2 + y**2 + 2*x*y, x**2 - 2], extension=sqrt(2)),
+                           [{x + sqrt(2), x + y}, {x + y, x - sqrt(2)}])
+
+    assert_equal_unordered(factor_system([x ** 2 + 1, y ** 2 + 1], gaussian=True),
+                           [{y + I, x - I}, {x - I, y - I}, {y + I, x + I}, {y - I, x + I}])
+
+    assert_equal_unordered(factor_system([x ** 2 + 1, y ** 2 + 1], domain=QQ_I),
+                           [{y + I, x - I}, {x - I, y - I}, {y + I, x + I}, {y - I, x + I}])
+
+    assert_equal_unordered(factor_system([0]), [set()])
+    assert_equal_unordered(factor_system([1]), [])
+    assert_equal_unordered(factor_system([0 , x]), [{x}])
+    assert_equal_unordered(factor_system([1, 0, x]), [])
+    assert_equal_unordered(factor_system([x**4 - 1, y**6 - 1]),
+                           [{x**2 + 1, y - 1}, {x**2 + 1, y + 1}, {x**2 + 1, y**2 + y + 1},
+                            {x**2 + 1, y**2 - y + 1}, {x - 1, y - 1}, {y - 1, x + 1},
+                            {x - 1, y**2 + y + 1}, {x + 1, y**2 + y + 1}, {x - 1, y + 1},
+                            {x - 1, y**2 - y + 1}, {x + 1, y + 1}, {x + 1, y**2 - y + 1}])
```

Comment:
When we have many repeating tests like this we can use parametrised tests. Then we can collect together all of these examples and use the same examples for all functions. See e.g.:
https://github.com/sympy/sympy/blob/614e48c61bfe8153c7293d1cff75544522668516/sympy/polys/matrices/tests/test_nullspace.py#L177-L188